modules/stochastic_tools/include/state_sim/model/StateSimModel.h

```diff
@@ -0,0 +1,58 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMMODEL_H
+#define STATESIMMODEL_H
+
+#include <string>
+#include <vector>
+#include "StateSimEvTimeBased.h"
+#include "StateSimEvConditionBased.h"
+#include "StateExternalCouplingUO.h"
+#include "StateSimAllEvents.h"
+#include "StateSimAllStates.h"
+#include "StateSimAllDiagrams.h"
+#include "StateSimAllVariables.h"
+#include "StateSimAllActions.h"
```

Comment:
In your .h file:
```
class StateSimAllStates;
class StateSimAllDiagrams;
```

Then in your .C file
```
#include "StateSimAllStates.h"
#include "StateSimAllDiagrams.h"
```

This works anytime you only use the name of a type and the compiler doesn't need to know the full implementation. That means when you have pointers and refs to types declared in your header which is quite often.